deploy/compose/osie/lastmile.sh

```diff
@@ -47,44 +48,54 @@ hook_rename_files() {
 	mv "${src_initrd}" "${dest_dir}/initramfs-x86_64"
 }
 
-# main runs the functions in order to download, extract, and move helper scripts
-main() {
+# process_osie_files processes the OSIE files and moves them to the correct location
+process_osie_file() {
 	local url="$1"
-	local extract_dir="$2"
-	local source_dir="$3"
-	local dest_dir="$4"
-	local use_hook="$5"
-
-	local filename
-	if [ "${use_hook}" == "true" ]; then
-		filename="osie-hook"
-	else
-		filename="osie"
-	fi
+	local filename="$2"
+	local extract_dir="$3"
+	local source_dir="$4"
+	local dest_dir="$5"
+	local use_hook="$6"
 
 	if [ ! -f "${extract_dir}"/"${filename}".tar.gz ]; then
-		echo "downloading osie..."
+		echo "downloading osie(s)..."
 		osie_download "${url}" "${extract_dir}" "${filename}"
 	else
 		echo "osie already downloaded"
 	fi
 
 	if [ "${use_hook}" == "true" ]; then
-		if [ ! -f "${source_dir}"/vmlinuz-x86_64 ] && [ ! -f "${source_dir}"/initramfs-x86_64 ]; then
-			echo "extracting hook..."
-			hook_extract "${extract_dir}" "${source_dir}" "${filename}"
-		else
-			echo "hook files already exist, not extracting"
-		fi
+		echo "extracting hook..."
+		hook_extract "${extract_dir}" "${source_dir}" "${filename}"
```

Comment:
did you mean to get rid of the check if already extracted?

Reply:
yeah. This hinders switching between hook and osie as these files would need to be overwritten but wouldn't because of these checks.

Reply:
how about adding a stamp file like `${extract_dir}/.hook-successfully-extracted` and similar for osie. Then the `if` would also be simpler too. I don't want to force anyone to re-download hook/osie unnecessarily.

Reply:
this is not for the download, just the archive extracting. The downloads will still be checked before trying again. see [here](https://github.com/tinkerbell/sandbox/blob/2bf87ecd0e4e3f400929eedc56939f6130b3a83f/deploy/compose/osie/lastmile.sh#L60)

Reply:
oops, right. But then does mean that osie/hook is going to be re-extracted everytime we run sandbox? If not, then why were the already-extracted-checks around for in the first place.

Reply:
yeah, an extract will happen with every `docker-compose up`. I removed them because the check for extracting based on files inside the archive feels unnecessary and couples this basic script to knowing what's inside the archive. For Hook there are 2 archives with 2 files in each, but OSIE has a lot of files. I don't know if it's worth checking. In my tests, the extract is normally only a few seconds.

Reply:
you likely have a fast ssd/nvme machine right? IIRC this was put in because extracting osie over and over was very slow for some users and very much a waste of time. You don't need to know what the files are that exist in the archive if you add your own stamp file post-extraction and check for that.

Reply:
I originated this script a month ago with the idea to have it be as idempotent as possible, not to solve any slowness or other. I would like to see this pushed through so I have updated the PR to check based on the downloaded archive.

Reply:
hmm then I'm thinking of something else then. Thanks for that context. All good then!